business/checkers/k8sreferencegrants/namespace_checker.go

```diff
@@ -28,3 +34,27 @@ func (in NamespaceChecker) Check() ([]*models.IstioCheck, bool) {
 
 	return validations, len(validations) == 0
 }
+
+type SingleNamespaceChecker struct {
+	Client         client.Reader
+	ReferenceGrant k8s_networking_v1beta1.ReferenceGrant
+}
+
+func (in SingleNamespaceChecker) Check() ([]*models.IstioCheck, bool) {
+	validations := make([]*models.IstioCheck, 0)
+
+	for nsIndex, from := range in.ReferenceGrant.Spec.From {
+		ns := &corev1.Namespace{}
+		if err := in.Client.Get(context.TODO(), types.NamespacedName{Name: string(from.Namespace)}, ns); err != nil {
+			if errors.IsNotFound(err) {
+				validation := models.Build("k8sreferencegrants.from.namespacenotfound",
+					fmt.Sprintf("spec/from[%d]/namespace", nsIndex))
+				validations = append(validations, &validation)
+			} else {
+				log.Errorf("Unable to get namespace %s while validating ReferenceGrant %s: %s", string(from.Namespace), in.ReferenceGrant.Name, err)
+			}
+		}
+	}
+
+	return validations, len(validations) == 0
+}
```

Comment:
@hhovsepy there's probably a better way to do this but the only reason I implemented this separately from the other checker is that 1. We don't need all namespaces 2. The other checker forces you to pass ALL reference grants when you only need to validate a single one.

Reply:
There could a performance impact requesting Namespaces and other resources in a loop like this `in.Client.Get(context.TODO(), types.NamespacedName{Name: string(from.Namespace)}, ns);`

Reply:
`in.Client` is the cache `Reader` so it should be fast but it would be good to ensure this is the case.